java/mykola-melnyk/src/com/zoolatech/lecture5/tasks/_4/Task4.java

```diff
@@ -0,0 +1,46 @@
+package com.zoolatech.lecture5.tasks._4;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * Create a method that accepts a string, counts all unique non-space character occurrences and
+ * displays each character with an amount of occurrences (output should be sorted by a character
+ * value in ascending order). Also, add a couple of examples that use your method.
+ * Input: “Hello World”
+ * Output: H - 1
+ * W - 1
+ * d - 1
+ * e - 1
+ * l - 3
+ * o - 2
+ * r - 1
+ */
+
+public class Task4 {
+    private static void charOccur3(String string) {
+        TreeMap<Character, Integer> myMap = new TreeMap<>();
+        char[] chars = string.toCharArray();
+        for (char ch : chars) {
+            if (ch != ' ') {
+                if (myMap.containsKey(ch)) {
+                    myMap.put(ch, myMap.get(ch) + 1);
```

Comment:
method map.getOrDefault could simplify the condition

Reply:
Can you please help me with transforming this condition with getOrDefault?

Reply:
sure:
```
if (myMap.containsKey(ch)) {
    myMap.put(ch, myMap.get(ch) + 1);
} else {
    myMap.put(ch, 1);
}
```
can be replaced with:
```
myMap.put(ch, myMap.getOrDefault(ch, 0) + 1); 
```
if a key is missing - the method will return 0, otherwise - a current value for the key